csm_web/frontend/src/components/course/ManageLabelsModal.tsx

```diff
@@ -0,0 +1,214 @@
+                  key={label.id}
+                  label={label}
+                  onChangeRow={(name: string, description: string, showPopup: boolean) =>
+                    editRow(label.id, name, description, showPopup)
+                  }
+                  removeLabel={() => removeLabel(label.id)}
+                />
+              ))}
+            </ul>
+          </div>
+          <div className="manage-label-modal-buttons">
+            <input className="affinity-btn" type="button" onClick={handleNewRow} value="Add New Label" />
+            <input className="affinity-btn" type="submit" value="Save Labels" />
+          </div>
+        </div>
+      </form>
+    </Modal>
+  );
+};
+
+interface EditLabelRowProps {
+  label: Label;
+  onChangeRow: (name: string, description: string, showPopup: boolean) => void;
+  removeLabel: (id: number) => void;
+}
+
+export const EditLabelRow = ({ label, onChangeRow, removeLabel }: EditLabelRowProps) => {
+  const [editingLabels, setEditingLabels] = useState<boolean>(false);
+
+  const toggleEditingLabels = (e: React.MouseEvent<HTMLOrSVGElement, MouseEvent>) => {
+    e.stopPropagation();
+    setEditingLabels(!editingLabels);
+  };
+
+  const editValue = (type: string, value: any) => {
+    if (type == "name") {
+      onChangeRow(value, label.description, label.showPopup);
+    } else if (type == "description") {
+      onChangeRow(label.name, value, label.showPopup);
+    } else if (type == "showpopup") {
+      onChangeRow(label.name, label.description, value);
+    } else {
+      // should not get here
+      console.error(`Unknown attribute ${type}`);
+    }
+  };
+
+  function handleChange({ target: { name, value } }: React.ChangeEvent<HTMLInputElement>) {
+    editValue(name, value);
+  }
+
+  function handleCheck(e: any) {
+    editValue("showpopup", e.target.checked);
+  }
+
+  return (
+    <div>
+      {/* // if not editing labels.. */}
+      {!editingLabels && (
+        // begining of evenly spaced row
+        <div className="manage-labels-modal-row">
+          <div className="manage-labels-modal-col large-col">
+            {label.name === "" && <div>Name of label</div>}
+            {label.name !== "" && <div>{label.name}</div>}
+          </div>
+          <div className="manage-labels-modal-col large-col">
+            {label.description === "" && <div>Description of label</div>}
+            {label.description !== "" && <div>{label.description}</div>}
+          </div>
+          <div className="manage-labels-modal-col small-col">
+            <input type="checkbox" defaultChecked={label.showPopup} onChange={handleCheck} disabled={true} />
+          </div>
+          <div className="manage-labels-modal-col small-col">
+            <Pencil className="icon matcher-assignment-section-times-edit-icon" onClick={toggleEditingLabels} />
+            <Trash className="icon matcher-assignment-section-times-edit-icon" onClick={removeLabel} />
+          </div>
```

Comment:
Add a hover effect for these 2 icons to make it more obvious that they can be interacted with. I would recommend using --csm-neutral for the default icon color and --csm-neutral-darkened when hovered over.